domain/model/state/state_test.go

```diff
@@ -1103,3 +1103,58 @@ func (m *stateSuite) TestGetModelByName(c *gc.C) {
 		OwnerName: m.userName,
 	})
 }
+
+// TestCleanupBrokenModel tests that when creation of a model fails (it is not
+// activated), and the user tries to recreate the model with the same name, we
+// can successfully clean up the broken model state and create the new model.
+// This is a regression test for a bug in the original code, where State.Create
+// was unable to clean up all the references to the original model.
```

Comment:
Can you please add the lp bug number to the comment so future readers can refer back.